test/parser/oracle.spec.ts

```diff
@@ -0,0 +1,193 @@
+import { parse } from '../../src/parser';
+import { expect } from 'chai';
+
+describe('Parser for oracle', () => {
+  describe('Given a CASE Statement', () => {
+    it('should parse a simple case statement', () => {
+      const sql = `SELECT CASE WHEN a = 'a' THEN 'foo' ELSE 'bar' END CASE from table;`;
+      const result = parse(sql, false, 'oracle');
+      expect(result.body.length).to.eql(1);
+    });
+  });
+
+  describe('given an anonymous block with an OUT pram', () => {
+    it('should treat a simple block as a single query', () => {
+      const sql = `BEGIN
+          SELECT
+            cols.column_name INTO :variable
+          FROM
+            example_table;
+        END`;
+      const result = parse(sql, false, 'oracle');
+      expect(result.body[0].type).to.eq('ANON_BLOCK');
+      expect(result.body[0].start).to.eq(0);
+      expect(result.body[0].end).to.eq(119);
+      expect(result.body.length).to.eql(1);
+    });
+
+    it('should easily identify two blocks', () => {
+      const sql = `BEGIN
+          SELECT
+            cols.column_name INTO :variable
+          FROM
+            example_table;
+        END;
+
+        BEGIN
+          SELECT
+            cols.column_name INTO :variable
+          FROM
+            example_table;
+        END
+        `;
+      const result = parse(sql, false, 'oracle');
+
+      expect(result.body[0].type).to.eq('ANON_BLOCK');
+      expect(result.body[0].start).to.eq(0);
+      expect(result.body[0].end).to.eq(120);
+      expect(result.body.length).to.eql(2);
```

Comment:
I think this should be the first assert where if there were no statements found, then we'd get an `undefined` error for a test failure, which isn't as meaningful imo.

Reply:
Yes I reorganized this incorrectly, moving the other way around